docs/superpowers/plans/2026-06-13-wire-onconfigreload.md

```
@@ -0,0 +1,1737 @@
+
+**Architecture:** Add a reusable core `BeanPostProcessor` extension point (mirroring the existing `CustomInjectorRegistry` factory). In `config-spigot`, an `OnConfigReloadProcessor` (a `BeanPostProcessor`) scans each bean's real class for `@OnConfigReload` methods and an `OnConfigReloadBinder` registers reload listeners on the config manager. Registered from `ConfigConfiguration` via a `BeanPostProcessorRegistryCustomizer` bean. A `ConfigRegistry` guard rejects the annotation on config POJOs.
+
+**Tech Stack:** Java 21, Maven (multi-module), JUnit 5, Mockito, Lombok. Spec: `docs/superpowers/specs/2026-06-13-wire-onconfigreload-design.md`.
+
+---
+
+## Conventions for this plan
+
+- **License header:** every NEW `.java` file must begin with the project's standard MIT license header. Copy it verbatim from any sibling source file (e.g. `core/.../postprocessor/BeanPostProcessor.java`). Code blocks below omit it for brevity — do not skip it in the actual file.
+- **Maven runner (Windows, path has spaces):** use the cached binary. Define once per shell:
+  ```bash
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)